internal/graphql/resolver/pattern.go

```diff
@@ -0,0 +1,39 @@
+package resolver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/layer5io/meshery/handlers"
+	"github.com/layer5io/meshery/internal/graphql/model"
+	"github.com/layer5io/meshery/models"
+)
+
+func (r *Resolver) fetchPatterns(ctx context.Context, provider models.Provider, selector model.PageFilter) (*model.PatternPageResult, error) {
+	tokenString := ctx.Value("token").(string)
+	user := ctx.Value(models.UserCtxKey).(*models.User)
+	prefObj := ctx.Value(models.PerfObjCtxKey).(*models.Preference)
+
+	resp, err := provider.GetMesheryPatterns(tokenString, selector.Page, selector.PageSize, *selector.Search, *selector.Order)
+
+	if err != nil {
+		r.Log.Error(err)
+		return nil, err
+	}
+
+	mc := handlers.NewContentModifier(tokenString, provider, prefObj, user.UserID)
+	err = mc.AddMetadataForPatterns(&resp)
+	if err != nil {
+		fmt.Println("Could not add metadata about pattern's current support ", err.Error())
```

Comment:
This error and the error on line 21... do these need to be MeshKit-style errors?

Reply:
Yeah.. but currently we used this error in the REST counterpart

Reply:
Fixed this